src/libostree/ostree-sysroot.c

```diff
@@ -812,6 +813,17 @@ parse_deployment (OstreeSysroot *self, const char *boot_link, OstreeDeployment *
       if (!glnx_fstat (deployment_dfd, &stbuf, error))
         return FALSE;
 
+      g_assert (self->run_ostree_metadata);
```

Comment:
I wouldn't use an assert here. You may run a newer ostree userspace on an older booted ostree which doesn't have the metadata in the ostree-booted file. We have a fallback anway for the dict not having the data, so just use the same if the file is missing.

Reply:
This is about the dict being constructed, not about the key present.  Note that previously `/run/ostree-booted` was just an empty file, and I believe creating a GVariantDict out of zero data is a perfectly cromulent empty dict.

Reply:
(Just tested by doing a `cp /dev/null /run/ostree-booted`, this works ok)